[System.Windows.Forms] Updated unit tests

filipnavara opened the pull request:
This is work on top of the https://github.com/mono/mono/pull/7401 pull request to fix and enable more unit tests.

akoeplinger commented:
@filipnavara now that https://github.com/mono/mono/pull/7401 was merged could you please rebase this PR against master?

filipnavara commented:
Yes, of course.

akoeplinger commented on mcs/class/System.Windows.Forms/Test/System.Windows.Forms/DataBindingTests.cs:
```diff
@@ -219,7 +219,7 @@ public void TestDeleteRowBeforeCurrent ()
 			Assert.AreEqual (0, cm.Position);
 
-			Console.WriteLine (event_log);
+			//Console.WriteLine (event_log);
```
it'd be nice to use `DebugWriteLine` for these too

filipnavara replied:
At first I thought about doing that, but there are asserts just on the next line for each of these, so the value is verified. And in case the assert fails the value is printed anyway. Maybe it would be better to just remove it altogether?

akoeplinger replied:
Yeah let's remove it in that case.

akoeplinger commented on mcs/class/System.Windows.Forms/Test/System.Windows.Forms/MdiFormHandleTest.cs:
```diff
@@ -1098,16 +1098,10 @@ public void Show ()
 		{
 			SetUp ();
 			Assert.IsFalse (child1.IsHandleCreated, "A1");
-			child1.HandleCreated += new EventHandler (HandleCreated_WriteStackTrace);
```
instead of removing the event handler we could set a variable that we assert below to ensure the event gets raised (assuming there's no such test already).

filipnavara replied:
I wasn't sure what was the purpose of the test. Will do it as you suggest.

filipnavara commented:
Incorporated feedback. Also, after rebase there appeared a new error with the X11 backend, so I fixed it too.

The X11 error is caused by behavior of XplatUIX11.CreateWindow and can happen in two cases:

- When window messages are pumped using WaitForHwndMessage. This in turn results in pumping timer (or other) messages that close the very same window that is just being created. Win32 never pumps messages in this manner, so the problem is specific to this backend.
- Any of the synchronous window messages (such as WM_SHOWWINDOW) results in call to Form.Close.

In either of those cases the window handle is no longer valid and should not be used. The problem was introduced by https://github.com/mono/mono/pull/7243, but the root cause was actually present before and only hidden from plain sight.
